python/GafferUI/SpreadsheetUI.py

```diff
@@ -0,0 +1,1560 @@
+				}
+			)
+
+def __addColumn( spreadsheet, plug ) :
+
+	# We allow the name of a column to be overridden by metadata, so that NameValuePlug and
+	# TweakPlug can provide more meaningful names than "name" or "enabled" when their child
+	# plugs are added to spreadsheets.
+	columnName = Gaffer.Metadata.value( plug, "spreadsheet:columnName" ) or plug.getName()
+
+	columnIndex = spreadsheet["rows"].addColumn( plug, columnName )
+	valuePlug = spreadsheet["rows"]["default"]["cells"][columnIndex]["value"]
+	Gaffer.MetadataAlgo.copy( plug, valuePlug, exclude = "spreadsheet:columnName layout:* deletable" )
+
+	return columnIndex
+
+def __createSpreadsheet( plug ) :
+
+	spreadsheet = Gaffer.Spreadsheet()
+	__addColumn( spreadsheet, plug )
+	spreadsheet["rows"].addRow()
+
+	with Gaffer.UndoContext( plug.ancestor( Gaffer.ScriptNode ) ) :
+		plug.node().parent().addChild( spreadsheet )
+		plug.setInput( spreadsheet["out"][0] )
+
+	GafferUI.NodeEditor.acquire( spreadsheet )
+
+def __addToSpreadsheet( plug, spreadsheet ) :
+
+	with Gaffer.UndoContext( spreadsheet.ancestor( Gaffer.ScriptNode ) ) :
+		columnIndex = __addColumn( spreadsheet, plug )
+		plug.setInput( spreadsheet["out"][columnIndex] )
+
+def __addToSpreadsheetSubMenu( plug ) :
+
+	menuDefinition = IECore.MenuDefinition()
+
+	alreadyConnected = []
+	other = []
+	for spreadsheet in Gaffer.Spreadsheet.Range( plug.node().parent() ) :
+
+		connected = False
+		for output in spreadsheet["out"] :
+			for destination in output.outputs() :
+				if destination.node() == plug.node() :
+					connected = True
+					break
+			if connected :
+				break
+
+		if connected :
+			alreadyConnected.append( spreadsheet )
+		else :
+			other.append( spreadsheet )
+
+	if not alreadyConnected and not other :
+		menuDefinition.append(
+			"/No Spreadsheets Available",
+			{
+				"active" : False,
+			}
+		)
+		return menuDefinition
+
+	alreadyConnected.sort( key = Gaffer.GraphComponent.getName )
+	other.sort( key = Gaffer.GraphComponent.getName )
+
+	def addItem( spreadsheet ) :
+
+		menuDefinition.append(
+			"/" + spreadsheet.getName(),
+			{
+				"command" : functools.partial( __addToSpreadsheet, plug, spreadsheet )
+			}
+		)
+
+	if alreadyConnected and other :
+		menuDefinition.append( "/__ConnectedDivider__", { "divider" : True, "label" : "Connected" } )
+
+	for spreadsheet in alreadyConnected :
+		addItem( spreadsheet )
+
+	if alreadyConnected and other :
+		menuDefinition.append( "/__OtherDivider__", { "divider" : True, "label" : "Other" } )
+
+	for spreadsheet in other :
+		addItem( spreadsheet )
+
+	return menuDefinition
+
+def __prependSpreadsheetCreationMenuItems( menuDefinition, plugValueWidget ) :
+
+	plug = plugValueWidget.getPlug()
+	if not isinstance( plug, Gaffer.ValuePlug ) :
+		return
+
+	node = plug.node()
+	if node is None or node.parent() is None :
+		return
+
+	if plug.getInput() is not None or not plugValueWidget._editable() or Gaffer.MetadataAlgo.readOnly( plug ) :
+		return
```

Comment:
11f26ee. I'm skipping the spreadsheet the plug belongs to, but otherwise allowing spreadsheet cells (and row names) to be added to _other_ spreadsheets. There was a small faction who were a bit sad that we'd omitted cascading fall-throughs from the row matching rules, and were consoling themselves with spreadsheet inception as an alternative workflow.

Reply:
Cool, in that case, do you think we could maybe add in a divider, and perhaps move it below `Disable Cell`, it feels that that is a more useful top-of-the-menu option?

Reply:
I have added a divider, but I don't think "Disable Cell" should go above "Create Spreadsheet...". Agreed that it's more useful when the menu is empty at the top, but it's often not empty at the top, and "Create Expression..." should logically be alongside "Create Spreadsheet...". And "Disable Cell" does seem like a reasonable neighbour for "Lock".

Reply:
Perfect